cmd/clusterctl/client/cluster/components_test.go

```diff
@@ -148,15 +157,17 @@ func Test_providerComponents_Delete(t *testing.T) {
 				{object: corev1.ObjectReference{APIVersion: "v1", Kind: "Pod", Namespace: "ns2", Name: "pod3"}, deleted: false},                                         // this object is in another namespace, and should never be touched by delete
 				{object: corev1.ObjectReference{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "ClusterRole", Name: "ns1-cluster-role"}, deleted: true},              // cluster-wide provider components should be deleted
 				{object: corev1.ObjectReference{APIVersion: "rbac.authorization.k8s.io/v1", Kind: "ClusterRole", Name: "some-cluster-role"}, deleted: false},            // other cluster-wide objects should be preserved
+				{object: corev1.ObjectReference{APIVersion: "clusterctl.cluster.x-k8s.io/v1alpha3", Kind: "Provider", Name: "providerOne"}, deleted: false},             // providerInventory should be preserved
```

Comment:
Let use the GroupVersion from the clusterctl API instead of "clusterctl.cluster.x-k8s.io/v1alpha3", so it will be easier to bump our API version in future